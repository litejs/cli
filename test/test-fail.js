
describe
.it("should fail on timeout", function(assert) {
	assert.setTimeout(5)
})
.it("should fail on multiple ends", function(assert) {
	setTimeout(this.end, 6)
	assert.end()
})
.it("should fail on assertion after end", function(assert) {
	assert.end()
	assert.ok(1)
})
.it("should fail on invalid assertions", function() {
	this.equal(
		[1, 2]  // <-- notice a comma is missing after "]"
		[3]
	)
	.equal(1, 2)
	.ok(0)
	.throws(function(){})
	.end()
})


