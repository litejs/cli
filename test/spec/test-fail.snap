Unnamed TestSuite
  ✘ it should fail #1 [0/1]
  ✘ it should fail on timeout #2 [0/0]
  ✘ it should fail on wrong plan 1 #3 [0/0]
  ✔ it should fail on wrong plan 2 #4 [1/1]
  ✔ it should fail on multiple ends #5 [0/0]
  ✔ it should fail on assertion after end #6 [0/0]
  ✔ it should pass on promise resolve #7 [1/1]
  ✘ it should fail on promise reject #8 [0/0]
ReferenceError: SOME_UNDEFINED_VALIABLE is not defined
    at Function.<anonymous> (test/test-fail.js:41:2)
    at nextCase (test.js:270:20)
    at end (test.js:333:23)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
  ✘ it should fail on invalid test #9 [0/0]
  ✘ it should fail on invalid assertions #10 [0/7]
Error: Unable to swap "prop"
    at Mock.swap (test.js:461:45)
    at Function.<anonymous> (test/test-fail.js:67:7)
    at nextCase (test.js:270:20)
    at Function.end (test.js:333:23)
    at Function.<anonymous> (test/test-fail.js:61:3)
    at nextCase (test.js:270:20)
    at end (test.js:333:23)
    at nextCase (test.js:274:5)
    at end (test.js:333:23)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
  ✘ it should fail on read-only swap #11 [0/0]
  ✘ it should handle an error without a stack #12 [0/0]
  ✘ it should handle an array as a stack #13 [0/0]
---
  ✘ it should handle an array as a stack #13 [0/0]
Error: 1
---
  ✘ it should handle an error without a stack #12 [0/0]
Error
---
  ✘ it should fail on read-only swap #11 [0/0]
Error: Unable to swap "prop"
---
  ✘ it should fail on invalid assertions #10 [0/7]
Error: Assertion:1: equal
expected: undefined
actual:   undefined
Error: Assertion:2: equal
expected: "2"
actual:   <null>{"a":Circular,"toJSON":()}
Error: Assertion:3: equal
expected: /1/m
actual:   /1/
Error: Assertion:4: equal
expected: []
actual:   arguments[testCase(value,message,actual,op,expected),false]
Error: Assertion:5: equal
expected: [/1/,B(),{"a":Circular,"toJSON":()}]
actual:   [/1/,B{},<null>{"a":Circular,"toJSON":()}]
Error: Assertion:6: 0 is truthy
Error: Assertion:7: throws
---
  ✘ it should fail on invalid test #9 [0/0]
ReferenceError: SOME_UNDEFINED_VALIABLE is not defined
---
  ✘ it should fail on promise reject #8 [0/0]
Error: Reject
---
  ✘ it should fail on wrong plan 2 #4 [2/2]
Error: assertion after end
Error: ended multiple times
---
  ✘ it should fail on assertion after end #6 [1/1]
Error: assertion after end
---
  ✘ it should fail on wrong plan 1 #3 [0/0]
Error: planned 1 actual 0
---
  ✘ it should fail on timeout #2 [0/0]
Error: TIMEOUT: 1ms
---
  ✘ it should fail #1 [0/1]
Error: Assertion:1: equal
expected: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.."
actual:   ["aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa","aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa","aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa","aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa","aaaaaaaaaaaaaaa..]
...
# FAIL  tests 1, 2, 3, 6, 4, 8, 9, 10, 11, 12, 13
1..13
# pass  2/13 [2/10]
---
  ✘ it should fail on multiple ends #5 [0/0]
Error: ended multiple times
...
# FAIL  tests 5
1..13
# pass  1/13 [2/10]
